From f8a4de69e7c1f546a6b2139297ac69d4d2bcf70e Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Fri, 3 Mar 2017 18:50:41 +0200
Subject: [PATCH 1548/5242] MLK-14301: imx dts: Fix cpu regulator references

commit  785bace97281ac386c6cb694907c9777f17d4ee9 from
https://source.codeaurora.org/external/imx/linux-imx.git

On imx_4.9.y we no longer override the cpu arm/soc/pu-supply properties
to link to pmic directly but rather rely on the regulator core
propagating a set_voltage on a bypassed regulator upwards.

This change was already performance for imx6qdl, now do it for all the
boards using ldo-bypass.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi     |   11 ++++++++---
 arch/arm/boot/dts/imx6qp-sabresd.dts           |    4 ----
 arch/arm/boot/dts/imx6sx-14x14-arm2.dts        |   12 ++++++++++--
 arch/arm/boot/dts/imx6sx-sdb.dts               |   10 ++++++++++
 arch/arm/boot/dts/imx6sxscm-evb.dts            |   13 +++++++++++--
 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts   |   12 ++++++++++--
 arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts  |   12 ++++++++++--
 arch/arm/boot/dts/imx6ul-14x14-lpddr2-arm2.dts |   12 ++++++++++--
 arch/arm/boot/dts/imx6ul-9x9-evk.dts           |   12 ++++++++++--
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts  |   12 ++++++++++--
 arch/arm/boot/dts/imx6ull-9x9-evk.dts          |   12 ++++++++++--
 11 files changed, 99 insertions(+), 23 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi b/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi
index 485f67e..0dd82ea 100644
--- a/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi
+++ b/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi
@@ -132,9 +132,14 @@
 
 };
 
-&cpu0 {
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1c_reg>;
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1c_reg>;
+	regulator-allow-bypass;
 };
 
 &clks {
diff --git a/arch/arm/boot/dts/imx6qp-sabresd.dts b/arch/arm/boot/dts/imx6qp-sabresd.dts
index eef9104..ddd9b68 100644
--- a/arch/arm/boot/dts/imx6qp-sabresd.dts
+++ b/arch/arm/boot/dts/imx6qp-sabresd.dts
@@ -54,10 +54,6 @@
 	status = "disabled";
 };
 
-&cpu0 {
-	arm-supply = <&sw2_reg>;
-};
-
 &iomuxc {
 	imx6qdl-sabresd {
 		pinctrl_usdhc2: usdhc2grp {
diff --git a/arch/arm/boot/dts/imx6sx-14x14-arm2.dts b/arch/arm/boot/dts/imx6sx-14x14-arm2.dts
index cf9809f..addce59 100644
--- a/arch/arm/boot/dts/imx6sx-14x14-arm2.dts
+++ b/arch/arm/boot/dts/imx6sx-14x14-arm2.dts
@@ -133,11 +133,19 @@
 		792000  1175000
 		396000  1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
 &ecspi4 {
 	fsl,spi-num-chipselects = <1>;
 	cs-gpios = <&gpio7 4 0>;
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index 6372f56..6ce4db1 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -36,6 +36,16 @@
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
 &qspi2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi2>;
diff --git a/arch/arm/boot/dts/imx6sxscm-evb.dts b/arch/arm/boot/dts/imx6sxscm-evb.dts
index 51c76b4..a905b71 100644
--- a/arch/arm/boot/dts/imx6sxscm-evb.dts
+++ b/arch/arm/boot/dts/imx6sxscm-evb.dts
@@ -499,11 +499,20 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+
 &qspi2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi2_1>;
diff --git a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
index a19fd06..2746935 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
@@ -118,11 +118,19 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
 &ecspi1 {
 	fsl,spi-num-chipselects = <1>;
 	cs-gpios = <&gpio4 26 0>;
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts b/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts
index 7bd356c..d319f2ea 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts
@@ -29,11 +29,19 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1_reg>;
-	soc-supply = <&sw1_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1_reg>;
+	regulator-allow-bypass;
+};
+
 &gpc {
 	fsl,ldo-bypass = <1>; /* use ldo-bypass */
 };
diff --git a/arch/arm/boot/dts/imx6ul-14x14-lpddr2-arm2.dts b/arch/arm/boot/dts/imx6ul-14x14-lpddr2-arm2.dts
index 0ec9509..bd87a0b 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-lpddr2-arm2.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-lpddr2-arm2.dts
@@ -129,11 +129,19 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
diff --git a/arch/arm/boot/dts/imx6ul-9x9-evk.dts b/arch/arm/boot/dts/imx6ul-9x9-evk.dts
index 8d032f1..e6c5c0f 100644
--- a/arch/arm/boot/dts/imx6ul-9x9-evk.dts
+++ b/arch/arm/boot/dts/imx6ul-9x9-evk.dts
@@ -172,11 +172,19 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1c_reg>;
-	soc-supply = <&sw1c_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1c_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1c_reg>;
+	regulator-allow-bypass;
+};
+
 &csi {
 	status = "disabled";
 
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
index 0b877e2..98b5cfa 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
@@ -115,11 +115,19 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1a_reg>;
+	regulator-allow-bypass;
+};
+
 &csi {
 	status = "okay";
 
diff --git a/arch/arm/boot/dts/imx6ull-9x9-evk.dts b/arch/arm/boot/dts/imx6ull-9x9-evk.dts
index ab10a43..8e88f66 100644
--- a/arch/arm/boot/dts/imx6ull-9x9-evk.dts
+++ b/arch/arm/boot/dts/imx6ull-9x9-evk.dts
@@ -172,11 +172,19 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1c_reg>;
-	soc-supply = <&sw1c_reg>;
 	fsl,arm-soc-shared = <1>;
 };
 
+&reg_arm {
+	vin-supply = <&sw1c_reg>;
+	regulator-allow-bypass;
+};
+
+&reg_soc {
+	vin-supply = <&sw1c_reg>;
+	regulator-allow-bypass;
+};
+
 &csi {
 	status = "okay";
 
-- 
1.7.9.5

